### PR TITLE
Updating MCT default GPTQ arguments with Gumbel Rounding compatible values

### PR DESCRIPTION
**Pull request by ofirgo**
- Minor changes in default parameters to Keras GPTQ facade.
- Added a 'gumbel_scale' argument to gumbel softmax.

### CODE REVIEW

File: model_compression_toolkit/gptq/common/gptq_config.py [hunk 1]
```diff
@@ -88,7 +89,8 @@ def __init__(self,
                  norm_weights: bool = False,
                  quantizer_config: GumbelConfig = GumbelConfig(),
                  optimizer_quantization_parameter: Any = None,
-                 optimizer_bias: Any = None):
+                 optimizer_bias: Any = None,
+                 gumbel_norm: float = 0.5):
```
Comment (haihabi) [inline model_compression_toolkit/gptq/common/gptq_config.py:93]: rename gumbel_norm to gumbel_varinace or gumbel_scale (https://en.wikipedia.org/wiki/Gumbel_distribution)

File: model_compression_toolkit/gptq/common/gptq_config.py [hunk 2]
```diff
@@ -88,7 +89,8 @@ def __init__(self,
                  norm_weights: bool = False,
                  quantizer_config: GumbelConfig = GumbelConfig(),
                  optimizer_quantization_parameter: Any = None,
-                 optimizer_bias: Any = None):
+                 optimizer_bias: Any = None,
+                 gumbel_scale: float = 0.5):
```
Comment (reuvenperetz) [inline model_compression_toolkit/gptq/common/gptq_config.py:93]: Please use a constant for this

File: model_compression_toolkit/gptq/keras/quantization_facade.py [hunk 3]
```diff
@@ -59,7 +59,7 @@
     def get_keras_gptq_config(n_iter: int,
                               optimizer: OptimizerV2 = tf.keras.optimizers.Adam(learning_rate=LR_DEFAULT),
                               optimizer_rest: OptimizerV2 = tf.keras.optimizers.Adam(learning_rate=LR_REST_DEFAULT),
-                              loss: Callable = GPTQMultipleTensorsLoss(),
+                              loss: Callable = GPTQMultipleTensorsLoss(norm_loss=False),
```
Comment (reuvenperetz) [inline model_compression_toolkit/gptq/keras/quantization_facade.py:62]: Maybe we should change the default value in GPTQMultipleTensorsLoss init method to False?

File: model_compression_toolkit/gptq/keras/quantizer/gumbel_rounding/gumbel_softmax.py [hunk 4]
```diff
@@ -29,7 +29,8 @@ def sample_gumbel(shape, eps=1e-6) -> tf.Tensor:
     return -tf.math.log(-tf.math.log(u + eps) + eps)
 
 
-def gumbel_softmax(in_pi: tf.Tensor, in_tau: tf.Tensor, in_gumbel: tf.Tensor, eps: float = 1e-6, axis=0) -> tf.Tensor:
+def gumbel_softmax(in_pi: tf.Tensor, in_tau: tf.Tensor, in_gumbel: tf.Tensor, eps: float = 1e-6, axis=0,
+                   gumbel_scale: float = 1.0) -> tf.Tensor:
```
Comment (reuvenperetz) [inline model_compression_toolkit/gptq/keras/quantizer/gumbel_rounding/gumbel_softmax.py:33]: Why default values of gumbel_scale are different here and in GumbelConfig init? Can we use the same default value?

File: model_compression_toolkit/gptq/keras/quantizer/gumbel_rounding/symmetric_gumbel.py [hunk 5]
```diff
@@ -73,7 +73,8 @@ def __init__(self, num_bits: int,
                  gumbel_config: GumbelConfig,
                  quantization_axis: int = -1,
                  max_lsbs_change_map: dict = DefaultDict({}, lambda: 1),
-                 max_iteration: int = 10000):
+                 max_iteration: int = 10000,
+                 gumbel_scale: float = 1.0):
```
Comment (reuvenperetz) [inline model_compression_toolkit/gptq/keras/quantizer/gumbel_rounding/symmetric_gumbel.py:77]: Same question as above-mentioned

File: model_compression_toolkit/gptq/keras/quantizer/gumbel_rounding/symmetric_gumbel.py [hunk 6]
```diff
@@ -73,7 +73,8 @@ def __init__(self, num_bits: int,
                  gumbel_config: GumbelConfig,
                  quantization_axis: int = -1,
                  max_lsbs_change_map: dict = DefaultDict({}, lambda: 1),
-                 max_iteration: int = 10000):
+                 max_iteration: int = 10000,
```
Comment (reuvenperetz) [inline model_compression_toolkit/gptq/keras/quantizer/gumbel_rounding/symmetric_gumbel.py:76]: Please use a constant for max_iterations default value.

File: model_compression_toolkit/gptq/keras/quantization_facade.py [hunk 7]
```diff
@@ -92,14 +92,15 @@ def get_keras_gptq_config(n_iter: int,
             The configuration can be passed to :func:`~model_compression_toolkit.keras_post_training_quantization` in order to quantize a keras model using gptq.
 
         """
-        bias_optimizer = tf.keras.optimizers.SGD(learning_rate=LR_BIAS_DEFAULT)
-        optimizer_quantization_parameter = tf.keras.optimizers.Adam(learning_rate=LR_QUANTIZATION_PARAM_DEFAULT)
+        bias_optimizer = tf.keras.optimizers.SGD(learning_rate=LR_BIAS_DEFAULT, momentum=0.9)
+        optimizer_quantization_parameter = tf.keras.optimizers.SGD(learning_rate=LR_QUANTIZATION_PARAM_DEFAULT, momentum=0.9)
```
Comment (reuvenperetz) [inline model_compression_toolkit/gptq/keras/quantization_facade.py:96]: Please use a constant for momentum value.
Also, it may be worth considering taking all GPTQ constants to a dedicated file (in a different PR).

**Review by reuvenperetz — CHANGES_REQUESTED** [review]

File: model_compression_toolkit/gptq/keras/quantizer/gumbel_rounding/gumbel_softmax.py (hunk 4 above)
Comment (ofirgo) [reply model_compression_toolkit/gptq/keras/quantizer/gumbel_rounding/gumbel_softmax.py:33]: It is intentional, the default in GumbelConfig is according to the behavior we want a MCT user will get (this is the best value for this parameters according to the tests we ran).
In the function itself I prefer to keep a default neutral value which doesn't affect the computation unless the user passes a specific scaling value.
Hope that makes sense..

File: model_compression_toolkit/gptq/keras/quantizer/gumbel_rounding/symmetric_gumbel.py (hunk 5 above)
Comment (ofirgo) [reply model_compression_toolkit/gptq/keras/quantizer/gumbel_rounding/symmetric_gumbel.py:77]: Same.

**Review by reuvenperetz — APPROVED** [review]
